### DISCUSSION TITLE
Error: Cannot perform I/O on behalf of a different request. I/O objects (such as streams, request/response bodies, and others) created in the context of one request handler cannot be accessed from a different request's handler. This is a limitation of Cloudflare Workers which allows us to improve overall performance.

### DISCUSSION THREAD

**Discussion by hiphopbesen** [vercel/community#2678] [Help] [answered]
### Page to Investigate

https://vermietung-black.vercel.app/settings

### Steps to Reproduce

I have a middleware function to handle checking and reauthorizing users. I am using the pocketbase authRefresh function. 
it works fine on my machine, yet when i deploy on vercel i get the following error

Error: Cannot perform I/O on behalf of a different request. I/O objects (such as streams, request/response bodies, and others) created in the context of one request handler cannot be accessed from a different request's handler. This is a limitation of Cloudflare Workers which allows us to improve overall performance.

This is the function i am using

`const authCheck = async (request: NextRequest) => {
  //read cookie
  let cookie = request.cookies.get('authjwt')?.value
  pb.authStore.save(cookie, null);
  const isTokenVerified = pb.authStore.isValid && !!(await pb.collection('users').authRefresh().catch((e:any) =>{console.log(e); false}));
  return isTokenVerified
}`

weirdest thing is that it works sometimes.

**Comment by TheSoeren** [comment, marked as answer]
I found a solution which worked for me here: https://github.com/pocketbase/pocketbase/discussions/2185#discussioncomment-5549838

Basically you need to create a new pocketbase instance for every serversided request.
